pkg/secretless/config/config.go

```diff
@@ -190,11 +36,38 @@ func LoadFromCRD(crdConfig crd_api_v1.Configuration) (config Config, err error)
 }
 
 // Load parses a YAML string into a Config object.
-func Load(data []byte) (config Config, err error) {
-	if err = parseConfigFile(data, &config); err != nil {
-		err = fmt.Errorf("Unable to parse configuration: '%s'", err)
+func Load(data []byte) (config v1.Config, err error) {
+	versionStruct := &struct {
+		Version string `yaml:"version"`
+	}{}
+
+	if err = yaml.Unmarshal(data, versionStruct); err != nil {
+		err = fmt.Errorf("unable to load configuration: '%s'", err)
+		return
+	}
+
+	var configPointer *v1.Config
+	if versionStruct.Version == "" {
+		versionStruct.Version = "1"
+	}
```

Comment:
I think you can set this in the struct before unmarshalling so if the field is not present, it will be defaulted to the value you've created the struct with.